### PR TITLE
Reuse HEVC put/put_uni optimizations

### PR DESCRIPTION
**Pull request by QSXW**

### CODE REVIEW

**Comment by QSXW** [comment]
@nuomi2021

**Comment by nuomi2021** [comment]
👍, awesome!

File: libavcodec/x86/vvc/vvc_mc.asm [hunk 1]
```diff
@@ -0,0 +1,296 @@
+; /*
+; * Provide SIMD MC functions for VVC decoding
+; * Copyright (c) 2023-2024 Nuo Mi
```
Comment (nuomi2021) [inline libavcodec/x86/vvc/vvc_mc.asm:3]: we copy the code from dav1d, better mentioned da1vd author here.

; Copyright © 2021, VideoLAN and dav1d authors
; Copyright © 2021, Two Orioles, LLC
; All rights reserved.

File: libavcodec/h26x/h2656_inter_template.c [hunk 2]
```diff
@@ -0,0 +1,572 @@
+/*
```
Comment (nuomi2021) [inline libavcodec/h26x/h2656_inter_template.c:1]: h2656_inter mismatched with h2656_mc.asm
better to use h2656_mc_template.c if you do not want change asm name.

Comment (QSXW) [reply libavcodec/h26x/h2656_inter_template.c:1]: inter is better. I will update them to inter.

File: libavcodec/Makefile [hunk 3]
```diff
@@ -65,6 +65,7 @@ OBJS = ac3_parser.o                                                     \
```
Comment (nuomi2021) [inline libavcodec/Makefile:65]: Need to change the commit log to h2656_inte.asm too

File: libavcodec/hevcdsp_template.c [hunk 4]
```diff
@@ -27,6 +27,13 @@
 #include "hevcdsp.h"
 #include "h26x/h2656_sao_template.c"
 
+#define LUMA_EXTRA        QPEL_EXTRA
```
[the remark this answers is not in the document]
Comment (QSXW) [reply libavcodec/hevcdsp_template.c]: Yes. Defined it in the h2656_inter_template.c

**Comment by QSXW** [comment]
@nuomi2021 All comments were resolved and passed the conformance tests both on Windows and Linux. Please help review the latest change.

**Comment by nuomi2021** [comment]
@QSXW Excellent work!
Feel free to proceed with sending it upstream.

**Comment by nuomi2021** [comment]
98b7a2a130a6e642a34501dba45a1807f912c103's ASM code changed too many things. 
It may need a long time to review it upstream.  split it to
1. move to h2656 only
2. change code
3. rename
may help the review process

**Comment by nuomi2021** [comment]
merged to upstream. thank you @QSXW 

Please help prepare ALF part.

**Comment by nuomi2021** [comment]
close this since it's upstreamed
thank you @QSXW

**Pull request closed without merging**